### PR TITLE
fix(target_chains/fuel): fuel merkle verification

### PR DESCRIPTION
**Pull request by cctdaniel**

### CODE REVIEW

File: target_chains/fuel/contracts/pyth-contract/src/pyth_merkle_proof.sw [hunk 1]
```diff
@@ -37,9 +37,9 @@ fn node_hash(child_a: Bytes, child_b: Bytes) -> Bytes {
 
 pub fn validate_proof(
     encoded_proof: Bytes,
-    leaf_data: Bytes,
     ref mut proof_offset: u64,
     root: Bytes,
+    leaf_data: Bytes,
```
Comment (cctdaniel) [inline target_chains/fuel/contracts/pyth-contract/src/pyth_merkle_proof.sw:42]: drive-by: standardizing it to be the same order as the [solidity counterpart](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/contracts/contracts/libraries/MerkleTree.sol#L47-L52)

**Review by ali-bahjati — APPROVED** [review]
